docs/src/components/hardware-list.tsx

```diff
@@ -0,0 +1,156 @@
+import React from "react";
+
+import {
+  Board,
+  HardwareMetadata,
+  Interconnect,
+  Shield,
+} from "../hardware-metadata";
+
+interface HardwareListProps {
+  items: HardwareMetadata[];
+}
+
+function itemIds(item: HardwareMetadata) {
+  if (item.type == "board" || item.type == "shield") {
+    let nodes = (item.siblings || [item.id])
+      .map<JSX.Element | string>((id) => <code key={id}>{id}</code>)
+      .reduce(
+        (prev, curr, index) => [...prev, index > 0 ? " and " : "", curr],
+        [] as Array<string | JSX.Element>
+      );
+    return <span key={item.id}>{nodes}</span>;
+  } else {
+    return <code key={item.id}>{item.id}</code>;
+  }
+}
+
+function typeLabel(type: HardwareMetadata["type"]): string {
+  switch (type) {
+    case "board":
+      return "Board: ";
+    case "shield":
+      return "Shield: ";
+    case "interconnect":
+      return "Interconnect: ";
+  }
+}
+
+function HardwareLineItem({ item }: { item: HardwareMetadata }) {
+  return (
+    <li>
+      <a href={item.url}>{item.name}</a> ({typeLabel(item.type)} {itemIds(item)}
+      )
+    </li>
+  );
+}
+
+function mapInterconnect({
+  interconnect,
+  boards,
+  shields,
+}: {
+  interconnect?: Interconnect;
```

Comment:
Minor: would probably be easier to read if you broke this out into an interface.

Reply:
Indeed. Pulled this out into a new `InterconnectDetails` type.